[plugin] Incompatible with Maven 3.9, missing plexus-utils

When running setup-disco under maven 3.9 it crashes with a class not found exception. It seems, that it requires plexus-utils which was dropped with maven 3.9.0.
See release notes of Maven 3.9.0 https://maven.apache.org/docs/3.9.0/release-notes.html
> Maven 2.x was auto-injecting an ancient version of plexus-utils dependency into the plugin classpath, and Maven 3.x continued doing this to preserve backward compatibility. Starting with Maven 3.9, it does not happen anymore. This change may lead to plugin breakage. The fix for affected plugin maintainers is to explicitly declare a dependency on plexus-utils. The workaround for affected plugin users is to add this dependency to plugin dependencies until issue is fixed by the affected plugin maintainer. See [MNG-6965](https://issues.apache.org/jira/browse/MNG-6965).

Adding the depedendency manually (see Step to Reproduce) fixes the problem.

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Install most recent maven version and jdks-maven-plugin.
2. Use the following plugin configuration:

```xml
            <plugin>
                <groupId>org.jreleaser</groupId>
                <artifactId>jdks-maven-plugin</artifactId>
                <version>1.22.0</version>
                <executions>
                    <execution>
                        <id>copy-client-jres</id>
                        <phase>process-resources</phase>
                        <goals>
                            <goal>setup-disco</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <pkgs>
                        <pkg>
                            <name>windows</name>
                            <platform>windows-x86_64</platform>
                            <packageType>jre</packageType>
                            <version>${java.vm.version}</version>
                            <distribution>temurin</distribution>
                        </pkg>
                        <pkg>
                            <name>mac</name>
                            <archiveType>tar.gz</archiveType>
                            <platform>osx-x64</platform>
                            <packageType>jre</packageType>
                            <version>${java.vm.version}</version>
                            <distribution>temurin</distribution>
                        </pkg>
                        <pkg>
                            <name>mac-arm</name>
                            <archiveType>tar.gz</archiveType>
                            <platform>osx-aarch64</platform>
                            <packageType>jre</packageType>
                            <version>${java.vm.version}</version>
                            <distribution>temurin</distribution>
                        </pkg>
                        <pkg>
                            <name>linux</name>
                            <archiveType>tar.gz</archiveType>
                            <platform>linux-x64</platform>
                            <packageType>jre</packageType>
                            <version>${java.vm.version}</version>
                            <distribution>temurin</distribution>
                        </pkg>
                    </pkgs>
                </configuration>
            </plugin>
```
   3. It crashes with a ClassNotFoundException.

When adding the dependency manually to the plugin, it works:

```xml
            <plugin>
                <groupId>org.jreleaser</groupId>
                <artifactId>jdks-maven-plugin</artifactId>
                <version>1.22.0</version>
...
                <dependencies>
                    <!-- Must override this dependency, even though not declared (see MNG-6965) -->
                    <dependency>
                        <groupId>org.codehaus.plexus</groupId>
                        <artifactId>plexus-utils</artifactId>
                        <version>3.5.1</version>
                    </dependency>
                </dependencies>
          </plugin>
```

### Expected Behaviour

It should not crash.

### Actual Behaviour

It crashes with ClassNotFoundException.

### Environment Information

- **Operating System**: Windows 11
- **JReleaser Version:** 1.22.0
- **JDK Version:** 25

Output mvn version:
```
Apache Maven 3.9.11 (3e54c93a704957b63ee3494413a2b544fd3d825b)
Maven home: C:\Development\maven
Java version: 25.0.1, vendor: Eclipse Adoptium, runtime: C:\Development\Java\jdk-25
Default locale: de_DE, platform encoding: UTF-8
OS name: "windows 11", version: "10.0", arch: "amd64", family: "windows"
```

```
A required class was missing while executing org.jreleaser:jdks-maven-plugin:1.22.0:setup-disco: org/codehaus/plexus/util/io/CachingOutputStream
-----------------------------------------------------
realm =    plugin>org.jreleaser:jdks-maven-plugin:1.22.0
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
urls[0] = file:/F:/packages/maven/org/jreleaser/jdks-maven-plugin/1.22.0/jdks-maven-plugin-1.22.0.jar
urls[1] = file:/F:/packages/maven/org/jreleaser/jreleaser-disco-java-sdk/1.22.0/jreleaser-disco-java-sdk-1.22.0.jar
urls[2] = file:/F:/packages/maven/org/jreleaser/jreleaser-utils/1.22.0/jreleaser-utils-1.22.0.jar
urls[3] = file:/F:/packages/maven/org/jreleaser/jreleaser-logger-api/1.22.0/jreleaser-logger-api-1.22.0.jar
urls[4] = file:/F:/packages/maven/org/jreleaser/jreleaser-resource-bundle/1.22.0/jreleaser-resource-bundle-1.22.0.jar
urls[5] = file:/F:/packages/maven/org/bouncycastle/bcpg-jdk18on/1.83/bcpg-jdk18on-1.83.jar
urls[6] = file:/F:/packages/maven/org/bouncycastle/bcprov-jdk18on/1.83/bcprov-jdk18on-1.83.jar
urls[7] = file:/F:/packages/maven/org/bouncycastle/bcutil-jdk18on/1.83/bcutil-jdk18on-1.83.jar
urls[8] = file:/F:/packages/maven/org/slf4j/jcl-over-slf4j/2.0.17/jcl-over-slf4j-2.0.17.jar
urls[9] = file:/F:/packages/maven/org/commonmark/commonmark/0.21.0/commonmark-0.21.0.jar
urls[10] = file:/F:/packages/maven/org/commonmark/commonmark-ext-autolink/0.21.0/commonmark-ext-autolink-0.21.0.jar
urls[11] = file:/F:/packages/maven/org/nibor/autolink/autolink/0.10.0/autolink-0.10.0.jar
urls[12] = file:/F:/packages/maven/org/apache/httpcomponents/httpclient/4.5.14/httpclient-4.5.14.jar
urls[13] = file:/F:/packages/maven/commons-logging/commons-logging/1.2/commons-logging-1.2.jar
urls[14] = file:/F:/packages/maven/org/apache/httpcomponents/httpcore/4.4.16/httpcore-4.4.16.jar
urls[15] = file:/F:/packages/maven/commons-codec/commons-codec/1.20.0/commons-codec-1.20.0.jar
urls[16] = file:/F:/packages/maven/io/github/openfeign/feign-core/13.6/feign-core-13.6.jar
urls[17] = file:/F:/packages/maven/io/github/openfeign/feign-jackson/13.6/feign-jackson-13.6.jar
urls[18] = file:/F:/packages/maven/com/fasterxml/jackson/core/jackson-core/2.20.1/jackson-core-2.20.1.jar
urls[19] = file:/F:/packages/maven/com/fasterxml/jackson/core/jackson-databind/2.20.1/jackson-databind-2.20.1.jar
urls[20] = file:/F:/packages/maven/com/fasterxml/jackson/core/jackson-annotations/2.20/jackson-annotations-2.20.jar
urls[21] = file:/F:/packages/maven/org/twdata/maven/mojo-executor/2.4.1/mojo-executor-2.4.1.jar
urls[22] = file:/F:/packages/maven/org/codehaus/plexus/plexus-utils/3.0.24/plexus-utils-3.0.24.jar
urls[23] = file:/F:/packages/maven/org/codehaus/plexus/plexus-archiver/4.10.4/plexus-archiver-4.10.4.jar
urls[24] = file:/F:/packages/maven/org/codehaus/plexus/plexus-io/3.6.0/plexus-io-3.6.0.jar
urls[25] = file:/F:/packages/maven/commons-io/commons-io/2.20.0/commons-io-2.20.0.jar
urls[26] = file:/F:/packages/maven/org/apache/commons/commons-compress/1.28.0/commons-compress-1.28.0.jar
urls[27] = file:/F:/packages/maven/org/apache/commons/commons-lang3/3.18.0/commons-lang3-3.18.0.jar
urls[28] = file:/F:/packages/maven/io/airlift/aircompressor/0.27/aircompressor-0.27.jar
urls[29] = file:/F:/packages/maven/org/tukaani/xz/1.10/xz-1.10.jar
urls[30] = file:/F:/packages/maven/com/github/luben/zstd-jni/1.5.7-6/zstd-jni-1.5.7-6.jar
Number of foreign imports: 1
import: Entry[import  from realm ClassRealm[project>de.foursoft.fourtime:fourtime-distribution:3.4.1-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]]

-----------------------------------------------------

    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:150)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.lang.NoClassDefFoundError: org/codehaus/plexus/util/io/CachingOutputStream
    at org.codehaus.plexus.archiver.tar.TarUnArchiver.execute (TarUnArchiver.java:97)
    at org.codehaus.plexus.archiver.tar.TarUnArchiver.execute (TarUnArchiver.java:84)
    at org.codehaus.plexus.archiver.AbstractUnArchiver.extract (AbstractUnArchiver.java:142)
    at org.jreleaser.jdks.maven.plugin.JdkHelper.extractJdk (JdkHelper.java:187)
    at org.jreleaser.jdks.maven.plugin.JdkHelper.setupJdk (JdkHelper.java:95)
    at org.jreleaser.jdks.maven.plugin.SetupDiscoMojo.doExecute (SetupDiscoMojo.java:54)
    at org.jreleaser.jdks.maven.plugin.AbstractDiscoMojo.execute (AbstractDiscoMojo.java:71)
    at org.jreleaser.jdks.maven.plugin.SetupDiscoMojo.execute (SetupDiscoMojo.java:38)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.lang.ClassNotFoundException: org.codehaus.plexus.util.io.CachingOutputStream
    at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass (SelfFirstStrategy.java:42)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass (ClassRealm.java:225)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:210)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:205)
    at org.codehaus.plexus.archiver.tar.TarUnArchiver.execute (TarUnArchiver.java:97)
    at org.codehaus.plexus.archiver.tar.TarUnArchiver.execute (TarUnArchiver.java:84)
    at org.codehaus.plexus.archiver.AbstractUnArchiver.extract (AbstractUnArchiver.java:142)
    at org.jreleaser.jdks.maven.plugin.JdkHelper.extractJdk (JdkHelper.java:187)
    at org.jreleaser.jdks.maven.plugin.JdkHelper.setupJdk (JdkHelper.java:95)
    at org.jreleaser.jdks.maven.plugin.SetupDiscoMojo.doExecute (SetupDiscoMojo.java:54)
    at org.jreleaser.jdks.maven.plugin.AbstractDiscoMojo.execute (AbstractDiscoMojo.java:71)
    at org.jreleaser.jdks.maven.plugin.SetupDiscoMojo.execute (SetupDiscoMojo.java:38)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

--- a/gradle.properties
+++ b/gradle.properties
@@ -100,6 +100,7 @@ picocliVersion         = 4.7.7
 plexusVersion          = 4.0.2
 plexusAnnotationsVersion = 2.2.0
 plexusArchiverVersion  = 4.10.4
+plexusUtilsVersion     = 3.5.1
 pomcheckerVersion      = 1.14.0
 tikaVersion            = 2.9.2
 wiremockVersion        = 2.35.0

--- a/plugins/jdks-maven-plugin/jdks-maven-plugin.gradle
+++ b/plugins/jdks-maven-plugin/jdks-maven-plugin.gradle
@@ -43,7 +43,7 @@ dependencies {
     api project(':jreleaser-disco-java-sdk')
     api "org.twdata.maven:mojo-executor:$mojoExecutorVersion"
     api "org.codehaus.plexus:plexus-archiver:$plexusArchiverVersion"
-    api "org.codehaus.plexus:plexus-utils:3.5.1"
+    api "org.codehaus.plexus:plexus-utils:$plexusUtilsVersion"
 
     compileOnly "org.apache.maven:maven-core:$mavenVersion"
     compileOnly "org.apache.maven:maven-plugin-api:$mavenVersion"

